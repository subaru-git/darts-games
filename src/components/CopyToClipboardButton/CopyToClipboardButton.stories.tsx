import type { Meta, StoryObj } from '@storybook/react';
import CopyToClipboardButton from './CopyToClipboardButton';

const meta: Meta<typeof CopyToClipboardButton> = {
  component: CopyToClipboardButton,
  title: 'Components/CopyToClipboardButton',
};

export default meta;
type Story = StoryObj<typeof CopyToClipboardButton>;
export const Default: Story = {};
