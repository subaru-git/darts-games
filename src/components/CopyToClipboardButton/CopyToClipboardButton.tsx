import React, { FC, useState } from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import Button from '@/atoms/Button';

type CopyToClipboardButtonProps = {
  text: string;
  label: string;
};

const CopyToClipboardButton: FC<CopyToClipboardButtonProps> = ({ text, label }) => {
  const [copied, setCopied] = useState(false);
  return (
    <div>
      <CopyToClipboard
        text={text}
        onCopy={() => {
          setCopied(true);
          setTimeout(() => setCopied(false), 3000);
        }}
      >
        <Button color='blue'>{copied ? 'Copied' : label}</Button>
      </CopyToClipboard>
    </div>
  );
};

export default CopyToClipboardButton;
