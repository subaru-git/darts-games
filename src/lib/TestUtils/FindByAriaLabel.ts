export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
