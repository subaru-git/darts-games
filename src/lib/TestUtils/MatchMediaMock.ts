import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
