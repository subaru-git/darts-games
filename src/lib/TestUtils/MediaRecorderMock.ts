export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
