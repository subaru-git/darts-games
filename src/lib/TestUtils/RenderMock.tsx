import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
