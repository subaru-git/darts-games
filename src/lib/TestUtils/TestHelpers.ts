import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}