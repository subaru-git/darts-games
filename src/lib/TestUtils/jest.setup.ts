import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
