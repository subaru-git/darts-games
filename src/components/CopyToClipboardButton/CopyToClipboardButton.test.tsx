import CopyToClipboardButton from './CopyToClipboardButton';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<CopyToClipboardButton />);
  expect(container).toMatchSnapshot();
});
