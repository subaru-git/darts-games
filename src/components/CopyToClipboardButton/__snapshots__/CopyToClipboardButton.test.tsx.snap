// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div>
    <button
      class="text-blue-400 border-blue-400 bg-white hover:bg-blue-50 w-fit rounded-lg border p-2 text-sm font-bold subpixel-antialiased disabled:cursor-not-allowed md:px-3 md:text-base"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
